datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java

```diff
@@ -1089,6 +1090,24 @@ private void hardDeleteDatastream(Datastream ds, List<Datastream> activeStreams)
     _adapter.deleteDatastream(ds.getName());
   }
 
+  /**
+   * Invokes post datastream state change action of connector for given datastream.
+   * @param datastream the datastream
+   * @throws DatastreamException if fails to perform post datastream action
+   */
+  public void invokePostDataStreamStateChangeAction(final Datastream datastream) throws DatastreamException {
+    _log.info("Invoke post datastream state change action");
+    try {
+      final String connectorName = datastream.getConnectorName();
+      final ConnectorInfo connectorInfo = _connectors.get(connectorName);
+      connectorInfo.getConnector().postDatastreamStateChangeAction(datastream);
```

Comment:
don't pass by reference, as the object may get modified in the connector. We have seen a bug in the past where the connector messed with the datastream cache.